src/main/java/io/appium/java_client/internal/ElementMap.java

```diff
@@ -68,17 +69,19 @@ public String getPlatformOrAutomation() {
     }
 
     /**
-     * @param hasSessionDetails something that implements {@link io.appium.java_client.HasSessionDetails}.
-     * @return subclass of {@link io.appium.java_client.MobileElement} that convenient to current session details.
+     * @param platform is the mobile platform. See {@link MobilePlatform}.
+     * @param automation is the mobile automation type. See {@link AutomationName}
+     * @return subclass of {@link org.openqa.selenium.remote.RemoteWebElement} that convenient
+     *     to current session details.
      */
-    public static Class<? extends RemoteWebElement> getElementClass(HasSessionDetails hasSessionDetails) {
-        if (hasSessionDetails == null) {
+    public static Class<? extends RemoteWebElement> getElementClass(String platform, String automation) {
+        if (isBlank(platform) && isBlank(automation)) {
             return RemoteWebElement.class;
         }
-        ElementMap element = Optional.ofNullable(mobileElementMap.get(String
-                .valueOf(hasSessionDetails.getAutomationName()).toLowerCase().trim()))
+        ElementMap element = Optional.ofNullable(mobileElementMap.get(
+                String.valueOf(platform).toLowerCase().trim()))
```

Comment:
does it make any sense to use `String.valueOf` for values of type String?

Reply:
@mykola-mokhnach It may return `null` value